### DISCUSSION TITLE
Q: UI-components, checkbox

### DISCUSSION THREAD

**Discussion by albertmink** [SAP/open-ux-tools#1769] [Q&A]
I've encountered an unusual behaviour with the UI-component checkbox that I hope to get some guidance on.

The issue involves a checkbox that has been set to both "checked" and "disabled". Normally, a disabled checkbox shouldn't respond to any user interactions. However, in this case, when I hover over the checkbox, it's altering its frame color to blue.

My understanding was that the color transformation signifies an active element, which seems to contradict the 'disabled' state. Is this an expected behavior or could it be a potential issue?

I would appreciate any knowledge or insights about why this might be happening or how to further troubleshoot this unusual behaviour.

Thank you in advance for your help and support!

Albert

**Comment by tobiasqueck** [comment]
@815are

**Comment by 815are** [comment]
Hi @gpeuc 

I think it is bug and we should not highlight checkbox on hover when it is checked and disabled.
+There is two inconsistencies:
1. checkbox - `checked=true + disabled=true`. We hover and see highlight/interaction blue border;
2. checkbox - `checked=false + disabled=true`. We hover and do not see highlight/interaction blue border;

Also if we look original component from fluent-ui - https://developer.microsoft.com/en-us/fluentui#/controls/web/checkbox
then it does not highlight disabled checkboxes in both cases.

**Comment by gpeuc** [comment]
Yep, of course. Disabled elements should not signalise they are interactive. 

By the way, @815are could you please check if other disabled elements are totally inactive? Buttons, input fields, dropdowns, and so on. When disabled, they should be completely sleeping. 😴

**Comment by 815are** [comment]
@gpeuc 

I rechecked and did not find other elements, which has such bug/behaviour.
Except special handling for inputs/comboboxes done in scope of - https://github.com/SAP/open-ux-tools/issues/801
but here we do not highlight on hover, only interaction part is that we allow to focus element(+focus style)

fyi @gpeuc @albertmink 
Bug regarding hoverable disabled checkbox https://github.com/SAP/open-ux-tools/issues/1776
